client/extensions/zoninator/components/forms/zone-content-form/index.jsx

```diff
@@ -15,50 +13,67 @@ import FormButton from 'calypso/components/forms/form-button';
 import SectionHeader from 'calypso/components/section-header';
 import PostsList from './posts-list';
 
-import 'calypso/state/form/init';
-
-const form = 'extensions.zoninator.zoneContent';
-
-class ZoneContentForm extends PureComponent {
+class ZoneContentForm extends Component {
 	static propTypes = {
 		disabled: PropTypes.bool,
-		handleSubmit: PropTypes.func.isRequired,
 		label: PropTypes.string.isRequired,
 		onSubmit: PropTypes.func.isRequired,
 		requesting: PropTypes.bool,
 		submitting: PropTypes.bool.isRequired,
 		translate: PropTypes.func.isRequired,
 	};
 
-	save = ( data ) => this.props.onSubmit( form, data );
+	state = {
+		posts: [],
+	};
+
+	componentDidUpdate( prevProps ) {
+		if (
+			! this.props.requesting &&
+			prevProps.requesting &&
+			this.props.initialValues &&
+			! this.state.posts.length
+		) {
+			// eslint-disable-next-line react/no-did-update-set-state
+			this.setState( {
```

Comment:
I'm not particularly fond of this solution, and I'm happy to drop it for a "mount this component after critical data is loaded" one if y'all feel it's better.